controllers/vsl.go

```diff
@@ -113,50 +113,33 @@ func (r *DPAReconciler) ValidateVolumeSnapshotLocations(dpa oadpv1alpha1.DataPro
 		if vslSpec.Velero == nil {
 			return false, errors.New("snapshotLocation velero configuration cannot be nil")
 		}
-		vsl := velerov1.VolumeSnapshotLocation{
-			ObjectMeta: metav1.ObjectMeta{
-				// TODO: Use a hash instead of i
-				Name:      fmt.Sprintf("%s-%d", r.NamespacedName.Name, i+1),
-				Namespace: r.NamespacedName.Namespace,
-			},
-			Spec: *vslSpec.Velero,
-		}
 
 		// check for valid provider
 		if vslSpec.Velero.Provider != AWSProvider && vslSpec.Velero.Provider != GCPProvider &&
 			vslSpec.Velero.Provider != Azure {
 			r.Log.Info("Non-supported provider specified, might be a misconfiguration")
 
-			r.EventRecorder.Event(&vsl,
-				corev1.EventTypeWarning,
-				"VSL provider is invalid",
-				fmt.Sprintf("VSL provider %s is invalid, might be a misconfiguration", vslSpec.Velero.Provider),
-			)
+			return false, fmt.Errorf("DPA spec.snapshotLocations[%v].velero.provider %s is invalid: only AWS, GCP and Azure are supported", i, vslSpec.Velero.Provider)
```

Comment:
I would use variables inside error messages as much as possible

For example, creating these variables inside for loop (after line 112)
```go
vslYAMLPath := fmt.Sprintf("spec.snapshotLocations[%v]", i)
veleroVSLYAMLPath := vslYAMLPath + ".velero"
```
So if these paths update in the future, it is easy to update error messages (only one place, instead of every message individually)

And then changing this message to
```go
return false, fmt.Errorf("DPA %v.provider %s is invalid: only %v are supported", veleroVSLYAMLPath, vslSpec.Velero.Provider, getVSLPluglins())
```
(`getVSLPluglins` does not exist, created for the example)

What you think?

Reply:
ya, variable part is good.

Reply:
changes pushed. Thanks you